src/simulation.jl

```diff
@@ -339,6 +339,10 @@ end =#
         if(operator == :(-))
           operator = promote_arithmetic_map[operator]
         end
+        if(operator == :(^))
```

Comment:
There is a function that would let you replace this logic with

`get(promote_arithmetic_map, operator, operator)`

Reply:
The main reason why we run these symbol checks is due to different behaviors between regular operations and dotted operations between a matrix and some other inputs. For example, ```mat^3``` will do three matrix multiplications while ```mat.^3``` will do element-wise exponentiation. 

To be honest, the checks for the multiplication and the exponentiation are a bit too loose here (subtraction is fine I think). We've gotten away with it since we apply matrices as function calls, not through multiplication, but this should be addressed in a refactor.